src/scanner/RepositoryConfig.ts

```diff
@@ -1,6 +1,11 @@
+import { ServiceType } from '../detectors/IScanningStrategy';
+
 export interface RepositoryConfig {
   remoteUrl: string | undefined;
   host: string | undefined;
   protocol: string | undefined;
   baseUrl: string | undefined;
+  basePath?: string; // local path to root folder (whether to the tmp folder if dxs scans remote url or local root folder of the repo)
+  localScanning?: boolean; // dxs is running remotely or local
+  serviceType?: ServiceType | undefined;
```

Comment:
This should be in ScanningStrategyDetector only because we can't detect service type in RepositoryExplorer but in ScanningStrategyDetector. We need to hold this logic separately. Do not mix two different configurations.